glide-core/redis-rs/redis/src/commands/cluster_scan.rs

```diff
@@ -256,25 +289,33 @@ impl ScanState {
     async fn creating_state_without_slot_changes<C: ClusterInScan + ?Sized>(
         &self,
         connection: &C,
+        allow_non_covered_slots: bool,
     ) -> RedisResult<ScanState> {
-        let next_slot = self.get_next_slot(&self.scanned_slots_map).unwrap_or(0);
+        let mut scanned_slots_map = self.scanned_slots_map;
+        let next_slot = self.get_next_slot(&scanned_slots_map).unwrap_or(0);
         let new_address = if next_slot == END_OF_SCAN {
```

Comment:
you compare the slot to  END_OF_SCAN in some places and NUM_OF_SLOTS in others, I think you should use only one of them

Reply:
In the example function I suggested where you do get_next_node with `while slot < NUM_OF_SLOTS`, you can skip this check and only do:
```
let next_slot = self.get_next_slot(&scanned_slots_map).unwrap_or(0);
match connection
                .get_address_by_slot(next_slot, &mut scanned_slots_map, allow_non_covered_slots)
                .await? {
AllSlotsCompleted => return Ok(ScanState::create_finished_state()),
Address(addr) => ...
}